src/kaleidoscope/EventQueue.h

```diff
@@ -0,0 +1,105 @@
+// -*- mode: c++ -*-
+/* Kaleidoscope - Firmware for computer input devices
+ * Copyright (C) 2013-2019  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <Arduino.h>
+
+#include "kaleidoscope/Kaleidoscope.h"
+#include "kaleidoscope/KeyAddr.h"
+
+namespace kaleidoscope {
+
+// This class defines a generic event queue that stores both key press and
+// release events, recording the key address, a timestamp, and the keyswitch
+// state (press or release).
+template <uint8_t _max_length,
+          typename _Bitfield  = uint8_t,
+          typename _Timestamp = uint16_t>
+class EventQueue {
+
+  static_assert(_max_length <= 255,
+                "EventQueue error: _max_length must be less than 256!");
+  static_assert(_max_length <= (sizeof(_Bitfield) * 8),
```

Comment:
This second `static_assert` is IMHO already sufficient.

Reply:
Right. I'm not sure why I added the first one. I started out with just this one, and now that I look at it, the first one would always pass because `_max_length` is a `uint8_t`.